pkg/api/pin_bytes.go

```diff
@@ -33,8 +34,14 @@ func (s *server) pinBytes(w http.ResponseWriter, r *http.Request) {
 	}
 
 	if !has {
-		jsonhttp.NotFound(w, nil)
-		return
+		_, err := s.Storer.Get(r.Context(), storage.ModeGetRequest, addr)
```

Comment:
That would be possible, however the reason I left existing code, i.e. `store.Has` function call (and why I would write it with it if I was doing it from scratch), was because I wanted to avoid unnecessary disk read operations.

As the operation flow goes:
- in the "best" case we use `store.Has`, which just does `leveldb.Has` operation, which should only check if key exists
  - after that we would write/update other key, for actual pinning operation
- if we were to use `store.Get` we have 2 internal cases
  - `netstore` case, where we do not have chunk locally
    - this would be worst case, and has to be done as is, no going around this
    - chunk will be retrieved (if possible) and added to localstore
  - localstore case, in which case actual content for the chunk would be read into newly allocated byte slice
    - we have no use for the chunk content at all, and will produce heap usage which is discarded

In regards to "more disk read", I know that that is a bit of a stretch comment, especially knowing how LevelDB actually works (not to mention also file system internal caching and prefetching), however I do think it is valid one. At least having less allocation should be enough of a reason for having current code as-is.

Reply:
cool. you've convinced me :+1: